libparsec/crates/client/tests/unit/workspace_inbound_sync_monitor.rs

```diff
@@ -0,0 +1,436 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use std::collections::VecDeque;
+use std::sync::{Arc, Mutex};
+use std::u64;
+
+use libparsec_platform_async::prelude::*;
+use libparsec_tests_fixtures::prelude::*;
+use libparsec_types::prelude::*;
+
+use super::{inbound_sync_monitor_loop, InboundSyncManagerIO, IncomingEvent};
+use crate::event_bus::AnySpiedEvent;
+use crate::workspace::{
+    InboundSyncOutcome, WorkspaceGetNeedInboundSyncEntriesError, WorkspaceSyncError,
+};
+
+/*
+ * Important stuff
+ */
+
+enum InboundSyncMonitorEvent {
```

Comment:
nit: I'd move the allows at the enum level

Reply:
And maybe for the complexity one maybe having something like that could do the trick
```rust
type MaybeAConvenientBox<T> = Option<Box<dyn FnOnce(T) + Send + Sync + 'static>>;
```

Reply:
:+1: done.